src/zolltools/nemsis/locationcodes.py

```diff
@@ -1,55 +1,108 @@
+    """
+
+    mapping = None
+
+
 def get_mapping() -> dict:
-    """Returns the Y92 mapping"""
+    """
+    Returns the Y92 mapping. The first read will read the mapping from storage.
+
+    :returns: the location code mapping
+    """
 
+    log_prefix = "get_mapping"
+
+    start_time = time.perf_counter_ns()
+    if _MappingStorage.mapping is not None:
+        return _MappingStorage.mapping
     root = resources.files(zolltools)
     mapping_file = root.joinpath("nemsis", "data", "y92-mapping.pkl")
+    logger.debug("%s: identified mapping file: %s", log_prefix, repr(mapping_file))
     with mapping_file.open("rb") as file:
-        return pickle.load(file)
+        mapping = pickle.load(file)
+        _MappingStorage.mapping = mapping
+        end_time = time.perf_counter_ns()
+        logger.info(
+            "%s: mapping read from %s in %d ns",
+            log_prefix,
+            mapping_file,
+            end_time - start_time,
+        )
+        return mapping
 
 
-def to_description(code, default=None):
-    """Returns the description for the provided code
-    Raises a KeyError exception if the code does not
-    match the data dictionary for Y92 code listing
+def to_description(code, default=None, mapping=None):
+    """
+    Returns the description for the provided code
+
+    :param code: the code to get a description of
+    :param default: the value to return if a description is not found. If set to `None`,
+    a KeyError exception will be raised instead.
+    :param mapping: the mapping to use. If set to `None`, the function will use the mapping
+    provided in the module: `locationcodes.get_mapping()`
+    :returns: a description of the Y92 code, `code`
+    :raises KeyError: if the code does not match the data dictionary for Y92 code listing
     """
 
+    log_prefix = "to_description"
+
+    if mapping is None:
+        logger.debug("%s: mapping accessed with get_mapping", log_prefix)
+        mapping = get_mapping()
+
     if code == "7701001":
         return "Not Applicable"
     elif code == "7701003":
         return "Not Recorded"
 
     try:
-        description = get_mapping()[code]
+        description = mapping[code]
     except KeyError as error:
         if default is None:
+            logger.error(
+                "%s: code (%s) not found in mapping %s", log_prefix, code, repr(mapping)
+            )
             raise KeyError(
                 f"{code} is an invalid code."
                 "Not in the NEMSIS data dictionary or Y92 code listings."
             ) from error
-        return "not-found"
+        return default
 
     return description
 
 
 def _get_storage_dir() -> Path:
     """Return a Path pointing to the directory holding the groupings"""
 
-    return Path.cwd().joinpath(STORAGE_FOLDER_NAME)
+    storage_dir_path = Path.cwd().joinpath(STORAGE_FOLDER_NAME)
+    logger.debug(
+        "_get_storage_dir: storage directory identified as %s", storage_dir_path
+    )
```

Comment:
Remove this logging statement. It doesn't seem particularly useful. Make sure the path is reported where this private method is used.

Reply:
resolved by new changes